### DISCUSSION TITLE
Migrating existing October projects

### DISCUSSION THREAD

**Discussion by climao** [wintercms/winter#101] [General]
Do you plan to provide an automated way to switch existing October projects to Winter?

**Comment by LukeTowers** [comment]
@climao yes, see https://github.com/wintercms/cli/issues/6. In the meantime though it's already extremely easy to migrate to Winter CMS. 

# Composer Migration

Change

```json
"october/rain": "1.1.*",
"october/system": "1.1.*",
"october/backend": "1.1.*",
"october/cms": "1.1.*",
```
to
```json
"winter/storm": "dev-develop as 1.1.999",
"winter/wn-system-module": "dev-develop",
"winter/wn-backend-module": "dev-develop",
"winter/wn-cms-module": "dev-develop",
```
and run `composer update`.

You will also need to update **bootstrap/autoload.php**'s 
```$helperPath = __DIR__.'/../vendor/october/rain/src/Support/helpers.php';```
to
```$helperPath = __DIR__.'/../vendor/winter/storm/src/Support/helpers.php';```

# Web Installer / Marketplace migration

If you're using a web installer version of October and you update through the backend, you just need to add the following entry to your `config/cms.php` file and then check for updates in the backend like you would normally:

```php
'updateServer' => 'https://api.wintercms.com/marketplace',
```

>**Note:** This method doesn't work just yet, but will as soon as we launch our new website, which should be no later than the end of the week

**Reply by DigiGNOSIS** [reply]
Added the 'updateServer' value mentioned above into the cms.php on an October 472 installation, at this moment the backend updater still says "No new updates were found." Is this functionality still pending?

**Reply by LukeTowers** [reply]
Yes

**Comment by climao** [comment]
@LukeTowers Thanks. Great news!

I do have some projects that were always updated through the backend, so the Marketplace migration comes in handy.
I'll be looking forward to it.

Still related to this subject, is there a simple way to convert a web installer version of October to a composer-updated version of Winter, short of creating a new project and copy everything from the old one?

**Reply by LukeTowers** [reply]
See https://github.com/wintercms/docs/blob/main/help-using-composer.md#converting-from-basic-install. The key thing is to set `cms.disableCoreUpdates` to true to prevent the backend from updating over the composer install in the future. I'm still working on the marketplace API to deliver the updates, but I'm thinking that I'm going to have it include a modified composer.json file in the update file so that all people would have to do to switch from basic to composer in the future is swap the config item.
